src/app/talent/talent/talent.component.ts

```diff
@@ -38,24 +42,70 @@ export class TalentComponent extends AbstractComponent<Talent> {
 	}
 
 	public onFileChange(event) {
-		const reader = new FileReader();
-
-		this.file = event.target.files[0];
-		reader.readAsDataURL(event.target.files[0]);
-		reader.onload = () => {
-			this.originalImage = reader.result;
+		if (event.target.name === 'listingImage') {
+			this.listingImage = event;
+			this.listingFile = event.target.files[0];
+		} else {
+			this.profileImage = event;
+			this.profileFile = event.target.files[0];
```

Comment:
use 2 different functions here: one for listingImage and one for profileImage. you don't need to do so many checks